sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -78,6 +78,27 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     }
   }
 
+  private def toSQL(node: LogicalPlan, topNode: Boolean): String = {
```

Comment:
At least for the purpose of SQLBuilder, I think the capability to convert a child of an analyzed logical plan is important. You can convert the children plans of the original plan back to several SQLs and perform these SQLs on different RDBMS, for example.